proof_of_stake/src/tests/state_machine.rs

```diff
@@ -2265,50 +2280,60 @@ impl AbstractPosState {
+                        if start <= infraction_epoch {
+                            let slashes_for_this_unbond = self
+                                .validator_slashes
+                                .get(&validator)
+                                .cloned()
+                                .unwrap_or_default()
+                                .iter()
+                                .filter(|&s| {
+                                    start <= s.epoch
+                                        && s.epoch
+                                            + self.params.unbonding_len
+                                            + self
+                                                .params
+                                                .cubic_slashing_window_length
+                                            < infraction_epoch
+                                })
+                                .cloned()
+                                .fold(
+                                    BTreeMap::<Epoch, Decimal>::new(),
+                                    |mut acc, s| {
+                                        let cur =
+                                            acc.entry(s.epoch).or_default();
+                                        *cur += s.rate;
+                                        acc
+                                    },
+                                );
+                            tracing::debug!(
+                                "Slashes for this unbond: {:?}",
+                                slashes_for_this_unbond
                             );
-                        tracing::debug!(
-                            "Slashes for this unbond: {:?}",
-                            slashes_for_this_unbond
-                        );
-                        total_unbonded += compute_amount_after_slashing(
-                            &slashes_for_this_unbond,
-                            unbond_amount,
-                            self.params.unbonding_len,
-                            self.params.cubic_slashing_window_length,
-                        );
+                            total_unbonded += compute_amount_after_slashing(
+                                &slashes_for_this_unbond,
+                                unbond_amount,
+                                self.params.unbonding_len,
+                                self.params.cubic_slashing_window_length,
+                            );
+                        } else {
+                            recent_unbonds += unbond_amount.change();
+                        }
 
                         tracing::debug!(
                             "Total unbonded (epoch {}) w slashing = {}",
                             epoch,
                             total_unbonded
                         );
                     }
+                    sum_post_bonds += self
+                        .total_bonded
+                        .get(&validator)
+                        .unwrap()
```

Comment:
There are cases in which this panics when a validator doesn't yet have any stake but it's still in consensus set as it's not yet full with non-zero stake validators. I think we can just change this to `.and_then(|bonded| bonded.get(&Epoch(epoch)))` to get around it and then we default to 0 if it's not set

Reply:
done in [7b4e333](https://github.com/anoma/namada/pull/1520/commits/7b4e3335cb50994e163a85549e9238c84c7b239b)